packages/apple-pay-integration/e2e/ApplePaySessionCustomerStepMockObject.ts

```diff
@@ -0,0 +1,180 @@
+const addApplePaySessionToChrome = () => {
+    class ApplePaySessionCustomerStep {
+        version: number;
+        paymentRequest: ApplePayJS.ApplePayPaymentRequest;
+
+        constructor(version, paymentRequest) {
+            this.version = version;
+            this.paymentRequest = paymentRequest;
+        }
+
+        STATUS_SUCCESS = 1;
+        STATUS_FAILURE = 2;
+
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        supportsVersion(versionNumber) {
+            return true;
+        }
+
+        canMakePayments() {
+            return true;
+        }
+
+        canMakePaymentsWithActiveCard() {
+            return Promise.resolve(this.canMakePayments());
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        completePayment() {
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+        completeShippingContactSelection(update) {
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+        completeShippingMethodSelection(update) {
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+        completeMerchantValidation(response) {
+        }
+
+        begin() {
+            setTimeout(() => {
+                const validationEvent = {
+                    validationURL: 'https://www.example.com',
+                } as ApplePayJS.ApplePayValidateMerchantEvent
+
+                const shippingContactEvent = {
+                    shippingContact: {
+                        email: "xx@xx.com",
+                        familyName: 'xx',
+                        givenName: 'xx'
+                    }
+                } as unknown as ApplePayJS.ApplePayShippingContactSelectedEvent;
+
+                const shippingMethodSelectedEvent = {
+                    shippingMethod: {
+                        label: 'xx',
+                        detail: 'xx',
+                        amount: "xx",
+                        identifier: 'xx'
+                    }
+                } as ApplePayJS.ApplePayShippingMethodSelectedEvent
+
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
```

Comment:
What are you ignoring? Sorry, just to provide you with some context, we're against using `ts-ignore` unless there's a valid reason, for example, to work around an incorrect type definition provided by a third party library etc...

Reply:
```
Property 'onpaymentauthorized' does not exist on type 'ApplePaySessionPaymentStep'.
```

We don't want to mock the native Apple Pay methods so I kept these empty

Reply:
Ok, the compiler is actually correct as the method is not defined. So what you need to do is to provide a no-op callback method. That'd be overridden by the implementation we're testing. e.g.:
```ts
onpaymentauthorized: (event: ApplePayJS.ApplePayPaymentAuthorizedEvent) => void = () => {
    console.log('onpaymentauthorized');
};
```

Reply:
Cool, I've implemented the no-op methods and override the implementations with the methods from the `ApplePaySession`